aodndata/aatams/aatams_sattag_qc_dm.py

```diff
@@ -0,0 +1,46 @@
+"""The AATAMS_SATTAG_QC_DM handler."""
+import os
+
+from aodncore.pipeline import HandlerBase, PipelineFilePublishType, FileType
+from .aatams_sattag_qc_dm_schema import AatamsSattagQcDmSchema
+
+# AATAMS DM global constants
+AATAMS_SATTAG_QC_DM_BASE = "IMOS/AATAMS/AATAMS_SATTAG_QC_DM"
+
+
+def dest_path(filepath):
+    return os.path.join(AATAMS_SATTAG_QC_DM_BASE, os.path.basename(filepath))
+
+
+class AatamsSattagQcDmHandler(HandlerBase):
+    """The class for aatams sattag qc dm."""
+
+    def __init__(self, *args, **kwargs):
+        """Initialize the class by only accepting zip files."""
+        super(AatamsSattagQcDmHandler, self).__init__(*args, **kwargs)
+        self.allowed_extensions = [".zip"]
+        self.allowed_regexes = ["^.+_.+$"]
+        self.allowed_archive_path_regexes = [AATAMS_SATTAG_QC_DM_BASE + ".+\\.zip"]
+        self.allowed_dest_path_regexes = [AATAMS_SATTAG_QC_DM_BASE + ".+\\.csv"]
+        self.dest_path_function = dest_path
+        self.archive_input_file = True
+        self.archive_path_function = self.dest_path_function
+
+        # Initialize the schema class and set default validation method
+        self.schema = AatamsSattagQcDmSchema()
+        self.validation_call = self.schema.extensive_validation
+        # Use below to just validate the file names and csv headers
+        # self.validation_call = self.SCHEMA.quick_validation
+
+    def preprocess(self):
+        """Validate individual files within the zip."""
+        files_in_zip = [x.local_path for x in self.file_collection]
+        self.validation_call(files_in_zip)
+        self.file_collection.add(self.input_file)
+        self.file_collection.filter_by_attribute_id(
+            "file_type", FileType.ZIP,
+        ).set_publish_types(PipelineFilePublishType.ARCHIVE_ONLY)
```

Comment:
Again, this can be done more cleanly (no need to filter for input_file in the collection, the handler already has the property [input_file_object](https://github.com/aodn/python-aodncore/blob/841b1adf00f79f1378d99a9134413e4cce8641d8/aodncore/pipeline/handlerbase.py#L511)):
```suggestion
        self.input_file_object.publish_type = PipelineFilePublishType.ARCHIVE_ONLY
        self.file_collection.add(self.input_file_object)
```

Reply:
done

Reply:
actually, I'm keeping the original. Changing to the above generate an error in the test suite.
input_file_object is readonly and not mutating the object state.

Reply:
If you got an error, I suspect you had a typo in there. My suggestion is exactly what e.g. [`AbosHandler`](https://github.com/aodn/python-aodndata/blob/67ffb3038301aa104ba40970016e08ddb4bc69d9/aodndata/moorings/handlers.py#L117) does.
Yes, `input_file_object` is a read-only property of the handler, that's not what we're changing. _We're setting a property of that_ `input_file_object`.

Reply:
done.

I felt into the trap of only reading only the help. I don't agree with the way it's done but I understand the reason why (lack of publish_type args at PipelineFile)